app/helpers/katello/katello_urls_helper.rb

```diff
@@ -1,15 +1,36 @@
 require 'uri'
+
 module Katello
   module KatelloUrlsHelper
+    def foreman_settings_url(schema = nil)
+      if schema
+        Settings[:foreman_url].sub(%r|^.*:|, "#{schema}:")
```

Comment:
I think we need to change the regexp to `%r|^.*?:|`, otherwise and url with port included (such as `http://foreman.example.com:3000` would end up with `https:3000`

Reply:
non greedy, nice catch